app/src/main/java/com/futurumgame/base/serialization/AndroidFileHelper.java

```diff
@@ -0,0 +1,80 @@
+package com.futurumgame.base.serialization;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.Log;
+
+import androidx.annotation.RequiresApi;
+
+import com.futurumgame.base.MainActivity;
+import com.futurumgame.base.exceptions.FileTooLargeException;
+import com.futurumgame.base.gameinternals.GameRoutine;
+import com.futurumgame.base.util.Logger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
+public class AndroidFileHelper {
+
+    private static AndroidFileHelper singleton;
+
+    private final Context context;
+
+    private AndroidFileHelper(Context context) {
+        this.context = context;
+    }
+
+    private File getFile(String fileName){
+        return new File(context.getFilesDir(), fileName);
+    }
+
+    public static void init(MainActivity activity) {
+        if(singleton != null) {
+            return;
+        }
+        singleton = new AndroidFileHelper(activity);
+    }
+
+    public static byte[] read(String fileName) throws IOException {
+        File file = singleton.getFile(fileName);
```

Comment:
Singleton null check or at least call the init method beforehand.... Could theoretically be null

Reply:
can not be null due to code convention, that all singletons have to be instantiated at the start of the game

Reply:
See question above.